lib/miq_automation_engine/engine/miq_ae_method_service/miq_ae_service_model_base.rb

```diff
@@ -245,6 +246,13 @@ def initialize(obj)
       raise MiqAeException::ServiceNotFound, "#{ar_klass.name} Object [#{obj}] not found" if @object.nil?
     end
 
+    def show_url
+      default_url_options[:host] = MiqRegion.my_region.remote_ui_url
+      url_for(:controller => ActiveModel::Naming.singular(self.class.ar_base_model),
+              :action     => 'show',
+              :id         => @object.id)
```

Comment:
What is the expected result if we do not have a controller for the resource being returned?  My guess is we are creating a URL that would look valid but not actually exposed.  For example, I do not think Lan/Switches have their own controller.  Also, since automate will dynamically create a default service model for any ActiveRecord backed object it could be almost anything.

Is there a way to return a `nil` value if we know it will be invalid?

Reply:
@gmcculloug I don't think every vmdb object is going to be exposed, currently we have a need only for MiqRequest, VM and Service which the user can view externally. The decision to call this method is in the Automate methods, so either we can have a generic function that gets called selectively or have a constrained function that works on only a few vmdb objects, this PR   uses the former approach.